### PR TITLE
Add ability to scroll horizontally in the server table

### PR DESCRIPTION
**Pull request by abist**
This PR fixes https://github.com/microsoft/azuredatastudio/issues/6141

Following from the initial thread, keeping this setting based on the workbench setting.

Capture

### CODE REVIEW

**Comment by anthonydresser** [comment]
Whats our actual ux guidelines here? before we didn't do this because vscode doesn't do this in their viewlets either, so we were staying consistent with vscode in terms of you just resize the view if you want to see more.

**Comment by abist** [comment]
@anthonydresser The `vscode` setting is `workbench.list.horizontalScrolling`. Perhaps the server tree should respect the same setting?

**Comment by anthonydresser** [comment]
I remember that coming up, I don't remember if it does respect it or not.

File: src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts [hunk 1]
```diff
@@ -140,7 +141,9 @@ export class ServerTreeView extends Disposable implements IServerTreeView {
 				this._connectionManagementService.showConnectionDialog();
 			}));
 		}
-		this._tree = this._register(TreeCreationUtils.createRegisteredServersTree(container, this._instantiationService));
+
+		const horizontalScrollEnabled: boolean = this._configurationService.getValue(horizontalScrollingKey) || false;
```
Comment (anthonydresser) [inline src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts]: This won't react to settings changes will it?

Comment (abist) [reply src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts]: This setting requires a reload for the changes to show up (even for vscode trees)

**Comment by peterlukerow** [comment]
I'm guessing that since the status says "Merged" this will be release in an upcoming ADS release?
Any ideas on when we might get this bug fix?

**Comment by Charles-Gagnon** [comment]
@peterlukerow If you wanted to try it out ahead of time you can download the [Insiders Build](https://github.com/microsoft/azuredatastudio#try-out-the-latest-insiders-build-from-master). Otherwise this will be included in the next stable release, which is targeted for sometime in the middle of April.

**Comment by peterlukerow** [comment]
Thanks for the update @Charles-Gagnon 
I can wait for the official release.

**Pull request merged**